components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -542,62 +545,37 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
             log.debug(String.format("Listing users by filter: %s %s %s", node.getAttributeValue(), node.getOperation(),
                     node.getValue()));
         }
-        try {
-            // Apply filter enhancements for username and update expression node value.
-            applyFilterEnhancementsForUsername(node);
-
-            if ((carbonUM instanceof PaginatedUserStoreManager) && isPaginatedUserStoreAvailable()) {
 
-                // If the limit is not specified, list all the users.
-                if (limit <= 0) {
-                    userNames = getAllFilteredUsers(node, limit, offset, domainName);
-                } else {
-                    // Extract he domain name if the domain name is embedded in the filter attribute value.
-                    domainName = resolveDomainNameInAttributeValue(domainName, node);
-
-                    // Get filtered user names.
-                    userNames = getFilteredUsers(node, offset, limit, sortBy, sortOrder, domainName);
-                }
-            } else
-                userNames = filterUsersUsingLegacyAPIs(node, limit, offset);
-        } catch (NotImplementedException e) {
-            String errorMessage = String.format("System does not support filter operator: %s", node.getOperation());
-            throw new CharonException(errorMessage, e);
+        // Apply filter enhancements for username and update expression node value.
+        applyFilterEnhancementsForUsername(node);
```

Comment:
Since this method enhance the user name and update that in the node, shall we change the method name a little bit to reflect that? Ex: `applyFilterEnhancementsForUsernameInNode`.

Reply:
fixed:  79e0f61a639ffe10869df3f4c7d5bf5611e2b731